pkgs/unified_analytics/USAGE_GUIDE.md

```diff
@@ -108,37 +108,40 @@ analytics.setTelemetry(status);
 
 ## Displaying Consent Message to Users
 
-When a user first uses any tool with this package enabled, they
-will be enrolled into Analytics collection. It will be the responsiblity
-of the tool using this package to display the proper Analytics messaging
-and inform them on how to Opt-Out of Analytics collection if they wish. The
-package will expose APIs that will make it easy to configure Opt-In status.
+When a user first uses any tool with this package enabled, the tool using
+this package will need to ensure that the user has seen the consent message.
+The tool using this package should check with the `Analytics` instance
+by invoking the `shouldShowMessage` getter. When this getter returns
+`true`, this means that the user has not been enrolled into analytics
+collection yet. It is at this point that the tool using this package will
+invoke the `getConsentMessage` getter to return a string to share with the
+user (each tool will have their own method of displaying the message
+through cli stdout, popup modal, etc.). Once the message has been shown,
+the tool using this package will need to confirm to the `Analytics` instance
+that it has shown the message; it is at this point that the user has
+officially been onboarded to analytics collection.
 
-For this use case, it is best to use the `Analytics` static method `getConsentMessage`
-which requires passing in the `DashTool` currently using this package.
 
-Passing in the `DashTool` is necessary so that the message can include which
-tool is seeking consent.
 
 ```dart
-final String consentMessage = Analytics.getConsentMessage(
-  tool: DashTool.flutterTools
-);
+// Begin by initializing the class near the entrypoint
+final Analytics analytics = Analytics(...);
+
+// This conditional should always run; the first time it is run, this
+// will return true since the consent message has never been shown
+if (analytics.shouldShowMessage) {
```

Comment:
Is it worth maintaining code in this markdown doc that isn't tested? What about just linking to https://github.com/dart-lang/tools/blob/main/pkgs/unified_analytics/example/unified_analytics_example.dart ? I believe with github there's a way to have relative links in markdown that github figures out, and will thus link to the correct version of the source.

This way, in the future when we change any of these APIs, analysis will fail on the example app, and we'll be forced to update it. Then these docs will always be up to date.